### DISCUSSION TITLE
Transcription generation from an audio file

### DISCUSSION THREAD

**Discussion by Genai-labs** [open-webui/open-webui#5144] [Ideas]
It would be interesting to explore the possibility of uploading audio files and automatically generating their transcription. This would help extend the solution's functionality, particularly by facilitating the analysis and utilization of audio content (without relying on live audio recording).

**Comment by pgbezerra** [comment]
We can easily implement that.

And adding a new conditional here: https://github.com/open-webui/open-webui/blob/main/backend/open_webui/apps/rag/main.py#L1236-L1277

Using this parser: https://github.com/langchain-ai/langchain/blob/de70a64e3a914a71aa875195b8171f559e13e5ca/libs/community/langchain_community/document_loaders/parsers/audio.py#L356

Also, if you use Tika, probably it would work out of the box. I'll test this before trying to change the backend-main code.

I don't know when I'll have time for it, but I need this feature too :D
So I hope I can do it soon.
